ci: failure in interface_usdt_utxocache.py

https://github.com/bitcoin/bitcoin/actions/runs/13501527489/job/37720897174?pr=31945:
```bash
UTXOCacheChange(outpoint=fab2396e3f726e1801b9e7303cf294783a3fd7e19ddc8f2dfa55e25b01575b3e:0, height=77, value=5000000000sat, is_coinbase=True) 
 test  2025-02-24T16:25:34.721000Z TestFramework (INFO): check that we successfully traced 2 adds and 1 spent 
 test  2025-02-24T16:25:34.721000Z TestFramework (ERROR): Assertion failed 
                                   Traceback (most recent call last):
                                     File "/home/runner/work/_temp/test/functional/test_framework/test_framework.py", line 135, in main
                                       self.run_test()
                                     File "/home/runner/work/_temp/build-asan/test/functional/interface_usdt_utxocache.py", line 149, in run_test
                                       self.test_add_spent()
                                     File "/home/runner/work/_temp/build-asan/test/functional/interface_usdt_utxocache.py", line 323, in test_add_spent
                                       compare_utxo_with_event(expected_utxo, actual_event)
                                     File "/home/runner/work/_temp/build-asan/test/functional/interface_usdt_utxocache.py", line 264, in compare_utxo_with_event
                                       assert_equal(utxo["txid"], bytes(event.txid[::-1]).hex())
                                     File "/home/runner/work/_temp/test/functional/test_framework/util.py", line 77, in assert_equal
                                       raise AssertionError("not(%s)" % " == ".join(str(arg) for arg in (thing1, thing2) + args))
                                   AssertionError: not(23b3dfef15c9b64445ffd44920adc48e7746aff1f03b59db3ed71936e90ff201 == d09ca39938b434b19a3338ef6b410071d85d4450dd7dc1f7db1bf7d95a53709a)
 test  2025-02-24T16:25:34.746000Z TestFramework (DEBUG): Closing down network thread 
```

Based on the logs just before the assert:

```log
TestFramework (INFO): handle_utxocache_add(): UTXOCacheChange(outpoint=d09ca39938b434b19a3338ef6b410071d85d4450dd7dc1f7db1bf7d95a53709a:0, height=201, value=4999968800sat, is_coinbase=False) 
TestFramework (INFO): handle_utxocache_add(): UTXOCacheChange(outpoint=23b3dfef15c9b64445ffd44920adc48e7746aff1f03b59db3ed71936e90ff201:0, height=201, value=2500031200sat, is_coinbase=True) 
TestFramework (INFO): handle_utxocache_spent(): UTXOCacheChange(outpoint=fab2396e3f726e1801b9e7303cf294783a3fd7e19ddc8f2dfa55e25b01575b3e:0, height=77, value=5000000000sat, is_coinbase=True) 
TestFramework (INFO): check that we successfully traced 2 adds and 1 spent 
[...]
AssertionError: not(23b3dfef15c9b64445ffd44920adc48e7746aff1f03b59db3ed71936e90ff201 == d09ca39938b434b19a3338ef6b410071d85d4450dd7dc1f7db1bf7d95a53709a)
```

It seems like we learned about `d09ca3..` and then about `fab239..` but expected the order to be the other way around. This makes sense, since `fab239..` is a coinbase UTXO which is added first. Unclear why the order was mixed in this case.

https://github.com/bitcoin/bitcoin/blob/0bb8a01810ec6f602b32d4677fa5004e12e0f538/test/functional/interface_usdt_utxocache.py#L319-L323

A simple solution could be to not depend on the order of the tracepoint event arrival. As long as all expected events are received, pass the test.

--- a/test/functional/interface_usdt_utxocache.py
+++ b/test/functional/interface_usdt_utxocache.py
@@ -265,14 +265,6 @@ class UTXOCacheTracepointTest(BitcoinTestFramework):
         actual_utxocache_adds = []
         actual_utxocache_spents = []
 
-        def compare_utxo_with_event(utxo, event):
-            """Compare a utxo dict to the event produced by BPF"""
-            assert_equal(utxo["txid"], bytes(event.txid[::-1]).hex())
-            assert_equal(utxo["index"], event.index)
-            assert_equal(utxo["height"], event.height)
-            assert_equal(utxo["value"], event.value)
-            assert_equal(utxo["is_coinbase"], event.is_coinbase)
-
         def handle_utxocache_add(_, data, __):
             event = ctypes.cast(data, ctypes.POINTER(UTXOCacheChange)).contents
             self.log.info(f"handle_utxocache_add(): {event}")
@@ -324,9 +316,28 @@ class UTXOCacheTracepointTest(BitcoinTestFramework):
 
         self.log.info(
             f"check that we successfully traced {EXPECTED_HANDLE_ADD_SUCCESS} adds and {EXPECTED_HANDLE_SPENT_SUCCESS} spent")
-        for expected_utxo, actual_event in zip(expected_utxocache_adds + expected_utxocache_spents,
-                                               actual_utxocache_adds + actual_utxocache_spents):
-            compare_utxo_with_event(expected_utxo, actual_event)
+
+        # Check that all expected tracepoints are recieved, but not the order they were recieved in.
+        # Tracepoint ordering is not strictly guaranteed, so this comparison avoids intermittent failures in the test.
+        def cache_event_to_key(event):
+            return (
+                bytes(event.txid[::-1]).hex(),
+                event.index,
+                event.height,
+                event.value,
+                event.is_coinbase
+            )
+
+        expected_add_keys = {(e["txid"], e["index"], e["height"], e["value"], e["is_coinbase"])
+                             for e in expected_utxocache_adds}
+        expected_spent_keys = {(e["txid"], e["index"], e["height"], e["value"], e["is_coinbase"])
+                              for e in expected_utxocache_spents}
+
+        actual_add_keys = {cache_event_to_key(e) for e in actual_utxocache_adds}
+        actual_spent_keys = {cache_event_to_key(e) for e in actual_utxocache_spents}
+
+        assert_equal(expected_add_keys, actual_add_keys)
+        assert_equal(expected_spent_keys, actual_spent_keys)
 
         bpf.cleanup()
 

